### PR TITLE
Update example formatting for better sphinx-gallery rendering

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

- Add plots to several examples
- Add `sphinx-gallery` section headings / description blocks to several examples
- Adjust plotting in some examples to avoid re-use of existing figures (which doesn't work well with `sphinx-gallery`) and use `plt.show()` more consistently for when examples are run from the command line
- More extensive revisions to `piston.py`, `mechanism_reduction.py`, `isentropic.py`, and `blowers_masel.py`
- Fix some issues with output files and missing `overwrite` flags when saving 1D flame data

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: samples/python/kinetics/mechanism_reduction.py [hunk 1]
```diff
@@ -25,42 +24,53 @@
 import cantera as ct
 import numpy as np
 import matplotlib.pyplot as plt
+from collections import namedtuple
```
Comment (bryanwweber) [inline samples/python/kinetics/mechanism_reduction.py]: If the goal here is attribute-based access to results, I think a `dataclass` would be more idiomatic. I know we're not necessarily trying to showcase Python best practices, but for the sake of <10 lines, I'd rather avoid `namedtuple`s. Here's some context for this idea, although it's not totally specific to the case here: https://snarky.ca/dont-use-named-tuples-in-new-apis/

File: samples/python/kinetics/shock_tube.py [hunk 2]
```diff
@@ -29,19 +29,21 @@
...
 
 import cantera as ct
 import matplotlib.pyplot as plt
 import numpy as np
 
-fig, ax = plt.subplots()
 file = 'example_data/ammonia-CO-H2-Alzueta-2023.yaml'
 models = {'Original': 'baseline', 'LMR-R': 'linear-Burke'}
-colours = ["xkcd:grey",'xkcd:purple']
+colours = {'Original': "xkcd:grey", 'LMR-R': 'xkcd:purple'}
```
Comment (bryanwweber) [inline samples/python/kinetics/shock_tube.py]:
```suggestion
colors = {'Original': "xkcd:grey", 'LMR-R': 'xkcd:purple'}
```

File: samples/python/kinetics/shock_tube.py [hunk 3]
```diff
@@ -61,18 +63,29 @@
...
           237.2,248.6,262.4,272.2,280.9],
     'X_H2O': [1.47E-03,1.59E-03,1.66E-03,1.78E-03,1.98E-03,2.06E-03,2.15E-03,2.22E-03,
               2.26E-03,2.30E-03,2.39E-03,2.38E-03,2.40E-03,2.42E-03,2.47E-03,2.53E-03,
               2.51E-03,2.50E-03,2.47E-03]
 }
+
+fig, ax = plt.subplots()
+
+for m, timeHistory in results.items():
+    ax.plot(timeHistory.t*1e6, timeHistory('H2O').X*100, color=colours[m], label=m)
```
Comment (bryanwweber) [inline samples/python/kinetics/shock_tube.py]:
```suggestion
    ax.plot(timeHistory.t*1e6, timeHistory('H2O').X*100, color=colors[m], label=m)
```

Comment (bryanwweber) [reply samples/python/kinetics/shock_tube.py]: For consistency with spelling everywhere else in Cantera

File: samples/python/reactors/combustor.py [hunk 4]
```diff
@@ -43,32 +46,34 @@
...
 exhaust = ct.Reservoir(gas)
 
+# %%
 # Use a variable mass flow rate to keep the residence time in the reactor
 # constant (residence_time = mass / mass_flow_rate). The mass flow rate function
 # can access variables defined in the calling scope, including state variables
-# of the Reactor object (combustor) itself.
-
-
+# of the `Reactor` object (combustor) itself.
```
Comment (bryanwweber) [inline samples/python/reactors/combustor.py:57]: In some places you have two backticks for monospace/API links, here's there's a single one. I didn't notice a pattern between the cases, so just asking for a double check :smile:

**Review by bryanwweber — COMMENTED** [review]
Some small suggestions, thanks @speth!

File: samples/python/reactors/combustor.py (hunk 4 above)
Comment (speth) [reply samples/python/reactors/combustor.py:57]: For ReST formatted documents, the single backtick utilizes the default Sphinx "role", which we have set to link to Python API objects, while double backticks is used for monospaced text. I double checked and I think all of the changes here are consistent with that usage.

Comment (bryanwweber) [reply samples/python/reactors/combustor.py:57]: Thanks for double checking!

**Review by ischoegl — APPROVED** [review]
Thanks, @speth!

**Pull request merged**